controllers/serveController.js

```diff
@@ -197,57 +90,47 @@ module.exports = {
-        serveFile(fileInfo, res);
-        sendGoogleAnalytics(method, headers, ip, originalUrl);
-        return fileInfo;
-      case SHOWLITE:
-        return db.Claim.resolveClaim(fileInfo.name, fileInfo.claimId)
-         .then(claimRecord => {
-           fileInfo['title'] = claimRecord.title;
-           fileInfo['description'] = claimRecord.description;
-           showFileLite(fileInfo, res);
-           return fileInfo;
-         })
-         .catch(error => {
-           logger.error('throwing serverOrShowAsset SHOWLITE error...');
-           throw error;
-         });
-      case SHOW:
-        return db.Claim
-          .getShortClaimIdFromLongClaimId(fileInfo.claimId, fileInfo.name)
-          .then(shortId => {
-            fileInfo['shortId'] = shortId;
-            return db.Claim.resolveClaim(fileInfo.name, fileInfo.claimId);
-          })
-          .then(resolveResult => {
-            logger.debug('resolve result >>', resolveResult.dataValues);
-            fileInfo['thumbnail'] = chooseThumbnail(resolveResult, DEFAULT_THUMBNAIL);
-            fileInfo['title'] = resolveResult.title;
-            fileInfo['description'] = resolveResult.description;
-            if (resolveResult.certificateId) { fileInfo['certificateId'] = resolveResult.certificateId };
-            if (resolveResult.channelName) { fileInfo['channelName'] = resolveResult.channelName };
-            showFile(fileInfo, res);
-            return fileInfo;
-          })
-          .catch(error => {
-            logger.error('throwing serverOrShowAsset SHOW error...');
-            throw error;
-          });
+  getLocalFileRecord (claimId, name) {
+    return db.File.findOne({where: {claimId, name}})
+      .then(file => {
+        if (!file) {
+          return NO_FILE;
+        }
+        return file.dataValues;
+      });
+  },
+  getClaimRecord (claimId, name) {
+    return db.Claim.findOne({where: {claimId, name}})
+        .then(claim => {
+          if (!claim) {
+            throw new Error('no record found in Claim table');
+          }
+          claim.dataValues.thumbnail = module.exports.chooseThumbnail(claim.dataValues, DEFAULT_THUMBNAIL);
+          claim.dataValues.fileExt = module.exports.determineFileExtensionFromContentType(claim.dataValues.contentType);
+          return claim.dataValues;
+        });
+  },
+  determineFileExtensionFromContentType (contentType) {
+    switch (contentType) {
+      case 'image/jpeg':
+        return 'jpeg';
+      case 'image/jpg':
+        return 'jpg';
+      case 'image/png':
+        return 'png';
+      case 'image/gif':
+        return 'gif';
+      case 'video/mp4':
+        return 'mp4';
       default:
-        logger.error('I did not recognize that method');
-        break;
+        logger.info('showing unknown file type as image/jpeg');
+        return 'jpeg';
+    }
+  },
+  chooseThumbnail (claimInfo, defaultThumbnail) {
+    if (!claimInfo.thumbnail || claimInfo.thumbnail.trim() === '') {
```

Comment:
I'd look at whether this trimming can be done in the model layer, rather than the controller (i.e. can we perform the TRIM in SQL or at least before the data makes it out of the model layer).

If you don't have to trim here, you can then just do things like:

`const thumb = claimInfo.thumbnail || DEFAULT_THUMBNAIL`

which is a lot cleaner and possibly obviates the need for the function entirely